### PR TITLE
SONARPY-507 Rule S3403: Identity operators should not be used with dissimilar types

### PR DESCRIPTION
**Pull request by pynicolas**

### CODE REVIEW

File: python-frontend/src/main/java/org/sonar/plugins/python/api/types/InferredType.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.python.api.types;
+
+public interface InferredType {
```
Comment (andrea-guarino-sonarsource) [inline python-frontend/src/main/java/org/sonar/plugins/python/api/types/InferredType.java]: As discussed, I think we can add `@Beta` on InferredType interface

File: python-frontend/src/main/java/org/sonar/python/tree/NameImpl.java [hunk 2]
```diff
@@ -91,4 +93,9 @@ public void setSymbol(Symbol symbol) {
   public void setUsage(Usage usage) {
     this.usage = usage;
   }
+
+  @Override
+  public InferredType type() {
+    return ((SymbolImpl) symbol).inferredType();
```
Comment (andrea-guarino-sonarsource) [inline python-frontend/src/main/java/org/sonar/python/tree/NameImpl.java]: I think that `symbol` might be null here.

File: python-frontend/src/main/java/org/sonar/plugins/python/api/types/InferredType.java (hunk 1 above)
Comment (pynicolas) [reply python-frontend/src/main/java/org/sonar/plugins/python/api/types/InferredType.java]: Done in 1f488f7.

File: python-frontend/src/main/java/org/sonar/python/tree/NameImpl.java (hunk 2 above)
Comment (pynicolas) [reply python-frontend/src/main/java/org/sonar/python/tree/NameImpl.java]: Fixed in b9f4ed6.

**Pull request merged**